crates/state/src/batch.rs

```diff
@@ -102,178 +128,156 @@ impl From<SignedBatchCheckpoint> for BatchCheckpoint {
+        l2_range: (L2BlockCommitment, L2BlockCommitment),
+        l2_blockid: L2BlockId,
+    ) -> Self {
+        Self {
+            epoch: checkpoint_idx,
+            l1_range,
+            l2_range,
+            l2_blockid,
+        }
     }
 
-    pub fn final_acc_pow(&self) -> u128 {
-        self.l1_pow_transition.1
+    pub fn epoch(&self) -> u64 {
+        self.epoch
     }
 
-    pub fn rollup_params_commitment(&self) -> Buf32 {
-        self.rollup_params_commitment
+    pub fn l2_blockid(&self) -> &L2BlockId {
+        &self.l2_blockid
     }
 
-    /// Creates a [`BootstrapState`] by taking the initial state of the [`BatchInfo`]
-    pub fn get_initial_bootstrap_state(&self) -> BootstrapState {
-        BootstrapState::new(
-            self.l1_range.0,
-            self.l1_transition.0,
-            self.l2_range.0,
-            self.l2_transition.0,
-            self.l1_pow_transition.0,
-        )
+    /// check for whether the l2 block is covered by the checkpoint
+    pub fn includes_l2_block(&self, l2_block_height: u64) -> bool {
+        let (_, last_l2_commitment) = self.l2_range;
+        if l2_block_height <= last_l2_commitment.slot() {
+            return true;
+        }
+        false
     }
 
-    /// Creates a [`BootstrapState`] by taking the final state of the [`BatchInfo`]
-    pub fn get_final_bootstrap_state(&self) -> BootstrapState {
-        BootstrapState::new(
-            self.l1_range.1 + 1, // because each batch is inclusive
-            self.l1_transition.1,
-            self.l2_range.1 + 1, // because each batch is inclusive
-            self.l2_transition.1,
-            self.l1_pow_transition.1,
-        )
-    }
-    /// check for whether the l2 block is covered by the checkpoint
-    pub fn includes_l2_block(&self, block_height: u64) -> bool {
-        let (_, last_l2_height) = self.l2_range;
-        if block_height <= last_l2_height {
+    /// check for whether the l1 block is covered by the checkpoint
+    pub fn includes_l1_block(&self, l1_block_height: u64) -> bool {
+        let (_, last_l1_commitment) = self.l1_range;
+        if l1_block_height <= last_l1_commitment.height() {
             return true;
         }
         false
     }
 }
 
-/// Initial state to bootstrap the proving process
+/// Represents the reference state against which batch transitions and proofs are verified.
 ///
-/// TODO: This needs to be replaced with GenesisState if we prove each Checkpoint
-/// recursively. Using a BootstrapState is a temporary solution
+/// NOTE/TODO: This state serves as the starting point for verifying a checkpoint proof. If we move
+/// towards a strict mode where we prove each checkpoint recursively, this should be replaced with
+/// `GenesisState`.
 #[derive(
     Clone, Debug, PartialEq, Eq, Arbitrary, BorshDeserialize, BorshSerialize, Deserialize, Serialize,
 )]
-pub struct BootstrapState {
-    pub start_l1_height: u64,
-    // TODO is this a blkid?
+pub struct CheckpointBaseState {
```

Comment:
Maybe should have `Commitment` in the name?  Since it's not actually the expanded state it's just a commitment to it.